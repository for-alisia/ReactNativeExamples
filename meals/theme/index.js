export default {
  colors: {
    primary: '#4a148c',
    accentColor: '#ff6f00',
    light: 'white'
  },
  randomColors: [
    '#f5428d',
    '#f54242',
    '#f5a442',
    '#f5d142',
    '#368dff',
    '#41d95d',
    '#46BFC7',
    '#439027',
    '#AD4CAD',
    '#47fced',
  ],
};
