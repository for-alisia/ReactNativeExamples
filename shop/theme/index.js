const theme = {
  colors: {
    primary: '#DC8080',
    primaryLight: '#ECBDBC',
    secondary: '#D8C496',
    secondaryLight: '#F8ECC9',
    accent: '#CA5454',
    accentDark: '#6B5344',
    light: '#F6F6F6',
    white: '#ffffff',
    grey: '#685E59',
    greyLight: '#ADA19B',
    dark: '#3C3735',
    yellow: '#FF9921',
  },

  fonts: {
    montserratReg: 'montserrat-regular',
    monserratBold: 'montserrat-bold',
    playfair: 'playfair',
    playfairBold: 'playfair-bold',
  },

  fontSize: {
    xs: 12,
    s: 16,
    m: 24,
    l: 32,
    xl: 102,
  },

  padding: {
    xs: 4,
    s: 8,
    m: 12,
    l: 16,
    xl: 24,
    xxl: 32,
  },

  margin: {
    xs: 4,
    s: 8,
    m: 12,
    l: 16,
    xl: 24,
    xxl: 32,
  },

  borderRadius: {
    xs: 4,
    s: 8,
    m: 16,
    l: 24,
  },

  activeOpacity: 0.75,
};

export default theme;
