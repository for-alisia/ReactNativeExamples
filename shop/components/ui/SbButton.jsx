// @ts-nocheck
// Dependencies
import React from 'react';
import { StyleSheet, View } from 'react-native';

// Components
import SbText from './SbText';
import SbTouchable from './SbTouchable';

// Theme
import theme from '../../theme';

const SbButton = (props) => {
  const { children, onPress, type, style, disabled } = props;

  const typeStyles = type || 'solid';
  const textStyles = typeStyles === 'solid' ? 'textSolid' : 'textOutlined';

  return (
    <SbTouchable onPress={!disabled ? onPress : () => {}}>
      <View
        style={{
          ...styles.container,
          ...styles[typeStyles],
          ...style,
          opacity: disabled ? 0.68 : 1,
        }}
      >
        <SbText style={{ ...styles.text, ...styles[textStyles] }}>{children}</SbText>
      </View>
    </SbTouchable>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingVertical: theme.padding.s,
    paddingHorizontal: theme.padding.xl,
    borderRadius: theme.borderRadius.l,
    alignSelf: 'center',
  },
  solid: {
    backgroundColor: theme.colors.primary,
  },
  outline: {
    backgroundColor: 'transparent',
    borderColor: theme.colors.primary,
    borderWidth: 1,
  },
  text: {
    textTransform: 'uppercase',
    fontSize: theme.fontSize.s,
    textAlign: 'center',
  },
  textSolid: {
    color: theme.colors.light,
  },
  textOutlined: {
    color: theme.colors.primary,
  },
});

export default SbButton;
