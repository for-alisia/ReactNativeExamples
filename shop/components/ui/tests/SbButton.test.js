// @ts-nocheck
// Dependencies
import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { toHaveStyle } from '@testing-library/jest-native';

// Component to test
import SbButton from '../SbButton';

// Theme
import theme from '../../../theme';

describe('SbButton tests', () => {
  // To use toHaveStyle
  expect.extend({ toHaveStyle });
  const testText = 'Hello';

  it('renders a button with correct text', () => {
    const { getByText, toJSON } = render(<SbButton>{testText}</SbButton>);
    const foundText = getByText(testText);
    expect(foundText.props.children).toEqual(testText);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correct styles by default', () => {
    const { getByTestId, getByText } = render(<SbButton>{testText}</SbButton>);
    const container = getByTestId('btn-container');
    const text = getByText(testText);

    expect(container).toBeTruthy();
    expect(text).toBeTruthy();
    expect(container).toHaveStyle({
      backgroundColor: theme.colors.primary,
    });
    expect(text).toHaveStyle({
      color: theme.colors.light,
    });
  });

  it('renders outlined styled button', () => {
    const { getByTestId, getByText, toJSON } = render(
      <SbButton type="outline">{testText}</SbButton>
    );
    const container = getByTestId('btn-container');
    const text = getByText(testText);

    expect(container).toBeTruthy();
    expect(text).toBeTruthy();

    expect(container).toHaveStyle({
      backgroundColor: 'transparent',
      borderColor: theme.colors.primary,
      borderWidth: 1,
    });

    expect(text).toHaveStyle({
      color: theme.colors.primary,
    });

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders a clickable button', () => {
    // Mock press function
    const onClickHandler = jest.fn();
    const { getByText } = render(<SbButton onPress={onClickHandler}>{testText}</SbButton>);
    const button = getByText(testText);
    fireEvent.press(button);
    expect(onClickHandler).toHaveBeenCalled();
  });

  it('renders disabled button', () => {
    // Mock press function
    const onClickHandler = jest.fn();
    const { getByText } = render(
      <SbButton onPress={onClickHandler} disabled>
        {testText}
      </SbButton>
    );
    const button = getByText(testText);
    fireEvent.press(button);
    expect(onClickHandler).not.toHaveBeenCalled();
  });
});
