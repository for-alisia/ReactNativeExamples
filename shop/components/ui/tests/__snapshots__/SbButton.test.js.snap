// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbButton UI Component renders a button with correct text 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#DC8080",
        "borderRadius": 24,
        "opacity": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 8,
      }
    }
    testID="btn-container"
  >
    <Text
      style={
        Object {
          "color": "#F6F6F6",
          "fontFamily": "montserrat-regular",
          "fontSize": 16,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`SbButton UI Component renders outlined styled button 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderColor": "#DC8080",
        "borderRadius": 24,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 8,
      }
    }
    testID="btn-container"
  >
    <Text
      style={
        Object {
          "color": "#DC8080",
          "fontFamily": "montserrat-regular",
          "fontSize": 16,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Hello
    </Text>
  </View>
</View>
`;
